pkg/vtpm/swtpm-vtpm/src/main.go

```diff
@@ -75,19 +82,27 @@ func makeDirs(dir string) error {
 	return nil
 }
 
+func readPidFile(pidPath string) (int, error) {
+	pidStr, err := os.ReadFile(pidPath)
+	if err == nil {
+		pid, err := strconv.Atoi(strings.TrimSpace(string(pidStr)))
+		if err == nil {
+			return pid, nil
+		}
+	}
+
+	return 0, fmt.Errorf("failed to read pid file: %w", err)
+}
+
 func getSwtpmPid(pidPath string, timeoutSeconds uint) (int, error) {
 	startTime := time.Now()
 	for {
 		if time.Since(startTime).Seconds() >= float64(timeoutSeconds) {
-			return 0, fmt.Errorf("timeout reached")
+			return readPidFile(pidPath)
```

Comment:
If swtpm is slow this will result in an ENOTEXIST error (I think) since the pidfile would not yet exist.
Better to transform such an error to "timed out after N seconds reading $pidPath